packages/oauth/src/reddit.ts

```diff
@@ -0,0 +1,227 @@
+import { get, post } from "./request.js";
+import type { Auth, GlobalUserAttributes, User } from "lucia-auth";
+import { generateState, GetAuthorizationUrlReturnType, OAuthConfig, OAuthProvider } from "./index.js";
+
+interface Configs extends OAuthConfig {
+	redirectUri: string;
+}
+
+class Reddit<A extends Auth> implements OAuthProvider {
+	constructor(auth: A, configs: Configs) {
+		this.auth = auth;
+		this.clientId = configs.clientId;
+		this.clientSecret = configs.clientSecret;
+		if (configs.scope) {
+			this.scope = ["identity", ...configs.scope];
+		} else {
+			this.scope = ["identity"];
+		}
+		this.redirectUri = configs.redirectUri;
+	}
+
+	private auth: A;
+	private clientId: string;
+	private clientSecret: string;
+	private scope: string[];
+	private redirectUri: string;
+
+	public getAuthorizationUrl = <State extends string | null | undefined = undefined>(
+		state?: State
+	): GetAuthorizationUrlReturnType<State> => {
+		const s = state ?? (typeof state === "undefined" ? generateState() : undefined);
+		const url = `https://www.reddit.com/api/v1/authorize?${new URLSearchParams({
+			client_id: this.clientId,
+			response_type: "code",
+			redirect_uri: this.redirectUri,
+			duration: "permanent",
```

Comment:
What does `duration` imply? Should it be mentioned in the docs?

Reply:
Takeing a look at the reddit doc. https://github.com/reddit-archive/reddit/wiki/OAuth2#authorization

> [duration] Indicates whether or not your app needs a permanent token. All bearer tokens expire after 1 hour. If you indicate you need permanent access to a user's account, you will additionally receive a refresh_token when acquiring the bearer token. You may use the refresh_token to acquire a new bearer token after your current token expires. Choose temporary if you're completing a one-time request for the user (such as analyzing their recent comments); choose permanent if you will be performing ongoing tasks for the user, such as notifying them whenever they receive a private message. The implicit grant flow does not allow permanent tokens.

From my understanding, we always need a permanent token for our use case. Therefore I don't see any necessity to add it to the docs.

Reply:
I think it should be mentioned so that the user knows what to look for in the reddit docs.

Reply:
Ok, I've added a line.